proposals/dao/tip_121c.ts

```diff
@@ -7,24 +7,58 @@ import {
   TeardownUpgradeFunc,
   ValidateUpgradeFunc
 } from '@custom-types/types';
-import { BigNumber } from 'ethers';
+import { expect } from 'chai';
+import { getImpersonatedSigner } from '@test/helpers';
+import { forceEth } from '@test/integration/setup/utils';
+import { expectApprox } from '@test/helpers';
 
-const toBN = BigNumber.from;
+const toBN = ethers.BigNumber.from;
 
 const fipNumber = 'tip_121c';
 
 let pcvStatsBefore: PcvStats;
 
+// Amount of FEI sudo() script mints to accounts[0], will be subtracted off
+const AMOUNT_FEI_MINTED_BY_E2E = toBN('10000000000000000000000000'); // 10M
+
+/////////////  Tribe Redeemer config
+// Circulating amount of TRIBE, which is redeemable for underlying PCV assets
+// TODO: Update with final circulating TRIBE figure
+const REDEEM_BASE = ethers.constants.WeiPerEther.mul(458_964_340);
+
+// Lido deposit balance, being withdrawn and sent to Tribe Redeemer
+const STETH_DEPOSIT_BALANCE = toBN('50296523674661485703301');
+const DAO_TIMELOCK_FOX_BALANCE = toBN('15316691965631380244403204');
+const DAO_TIMELOCK_LQTY_BALANCE = toBN('1101298805118942906652299');
+
+// User circulating Fei as determined by fei-tools.com for block 15589242
+const USER_CIRCULATING_FEI_AT_FIXED_BLOCK = ethers.constants.WeiPerEther.mul(58_708_166);
+
+// Minimum DAI transferred to Redeemer. Lower bound
+// TODO: Update with final numbers
+const MIN_REMAINING_DEPOSIT_DAI_FOR_REDEEMER = ethers.constants.WeiPerEther.mul(30_000_000);
+
 // Do any deployments
 // This should exclusively include new contract deployments
 const deploy: DeployUpgradeFunc = async (deployAddress: string, addresses: NamedAddresses, logging: boolean) => {
+  // SimpleFeiPSM
   const simpleFeiDaiPSMFactory = await ethers.getContractFactory('SimpleFeiDaiPSM');
   const simpleFeiDaiPSM = await simpleFeiDaiPSMFactory.deploy();
   await simpleFeiDaiPSM.deployed();
-  logging && console.log(`simpleFeiDaiPSM: ${simpleFeiDaiPSM.address}`);
+  console.log('Simple FEI PSM deployed to: ', simpleFeiDaiPSM.address);
+
+  // Tribe Redeemer
+  const tribeRedeemerFactory = await ethers.getContractFactory('TribeRedeemer');
+  const redeemedToken = addresses.tribe;
+  const tokensReceived = [addresses.steth, addresses.lqty, addresses.fox, addresses.dai];
+
+  const tribeRedeemer = await tribeRedeemerFactory.deploy(redeemedToken, tokensReceived, REDEEM_BASE);
+  await tribeRedeemer.deployTransaction.wait();
+  console.log('Tribe redeemer deployed to: ', tribeRedeemer.address);
 
   return {
-    simpleFeiDaiPSM
+    simpleFeiDaiPSM,
+    tribeRedeemer
   };
 };
 
```

Comment:
TODO: in setup() verify dripper and holding deposit paused + check reverts if try to allocateSurplus() or drip

Reply:
updated plan